test/Makefile

```diff
@@ -0,0 +1,78 @@
+comma := ,
+colon := :
+space :=
+space +=
+
+define colorecho
+	@tput setaf 6
+	@echo $1
+	@tput sgr0
+endef
+
+define capture-output
+	script $1 $2 > /dev/null || cat $1
+endef
+
+ROOT ?= ..
+VIRTUALENV_ROOT := $(ROOT)/venv
+PATH := $(VIRTUALENV_ROOT)/bin:$(PATH)
+SHELL = bash
+
+SRCDIR := $(ROOT)/job_definitions
+TEST_CACHE := .test_cache
+
+ANSIBLE_OUTDIR := $(TEST_CACHE)/job_builder_definitions
+JJB_OUTDIR := $(TEST_CACHE)/jenkins_job_definitions
+
+ifdef JOBS
+	JOBS_LIST := $(subst $(comma),$(space),$(JOBS))
+else
+	JOBS_LIST := $(patsubst %.yml,%,$(notdir $(wildcard $(SRCDIR)/*.yml)))
+endif
+
+SOURCES := $(foreach JOB,$(JOBS_LIST),$(SRCDIR)/$(JOB).yml)
+TARGETS := $(foreach JOB,$(JOBS_LIST),$(JJB_OUTDIR)/$(JOB).xml)
+
+test-job-templates: $(TARGETS)
+
+pre-populate-cache: $(ANSIBLE_OUTDIR)
+
+define process-templates
+	ansible-playbook \
+		--inventory="localhost," \
+		--extra-vars=@$(ROOT)/playbooks/roles/jenkins/defaults/main.yml \
+		--extra-vars=@$(ROOT)/test/fixtures/vars.yml \
+		--extra-vars="dest=$1" \
+		--extra-vars="items=[$(subst $(space),$(comma),$2)]" \
+		$(ROOT)/test/test_templates.yml
+endef
+
+define process-templates-quiet
+	$(call capture-output,$(@:.yml=).log,$(call process-templates,$1,$2))
+endef
+
+$(ANSIBLE_OUTDIR):
+	@mkdir -p $@
+	@$(call colorecho,'Detected first run; generating cache...')
+	@$(call process-templates,$@,$(SOURCES))
+	@$(eval ALREADY_RAN := true)
```

Comment:
Does it take a lot longer to rebuild the cache for each test run? Not sure we want it handing around 🤷‍♂️

Reply:
I timed it as being 2 minutes instead of 30 seconds, which I thought was a pain, which is why I added the caching. But Make magically deletes the intermediate files at the end if you run `make test` without generating the cache first so it doesn't take up too much disk space.